### ISSUE TITLE
Uninstalling a nuget package removes reference to a project with the same name

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/project-system#5238]
_This issue has a corresponding [ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/653221/uninstalling-a-nuget-package-removes-reference-to.html). Please vote and comment there to make sure your voice is heard._

---

Uninstalling a nuget package in the nuget package manager removes reference to a project with the same name

---

#### Andy Zivkovic [MSFT] on 7/19/2019, 02:49 PM

<p>I created a .NET Standard class library named “Newtonsoft.Json”, created a .NET Core console app, added a project reference to Newtonsoft.Json, added a package reference to Newtonsoft.Json then used the Visual Studio manage manager UI to uninstall the package, and the the project reference remained.</p>
<p>Can you provide more details?</p>

#### Nikolche Kolev [MSFT] on 7/29/2019, 11:53 AM

<p>It seems like in your scenario you are using packages.config based project.<br><br>Are you suggesting nuget removes the ProjectReference declaration? <br>Can you please provide more details about the scenario? <br><br>Please follow: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2019#when-further-information-is-needed-need-more-info">https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2019#when-further-information-is-needed-need-more-info</a> on how to provide more information. </p>

#### Visual Studio Feedback System on 7/30/2019, 10:57 AM

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

**Comment by davkean** [comment]
@etbyrd What are you blocked on?

**Comment by etbyrd** [comment]
I left a private comment asking for more specific information from the customer.

**Comment by ocallesp** [comment]
We have investigated and after further investigation we have determined that this is not a supported scenario. Our code is not designed to deal with two references having the same name.

In the old project, the alias Newtonsoft.Json is the same for the NuGet package and project name.
Making changes to support this scenario is too risky.

https://developercommunity.visualstudio.com/content/problem/653221/uninstalling-a-nuget-package-removes-reference-to.html
